agent/handlers/gql/schema_test.go

```diff
@@ -532,26 +1255,39 @@ func TestCreateSchema(t *testing.T) {
 				ecsClient.EXPECT().GetResourceTags(taskARN).Return(nil, errors.New("Mock Error")).AnyTimes()
 			} else if tc.name == "Tags Query" {
 				tagsQueryHelper(ecsClient)
-			} else {
+			} else if tc.containerByIdError {
+				state.EXPECT().TaskByID(fakeContainerId).Return(task, true).AnyTimes()
+				state.EXPECT().ContainerByID(fakeContainerId).Return(nil, false).AnyTimes()
+				state.EXPECT().ContainerMapByArn(taskARN).Return(containerNameToDockerContainer, true).AnyTimes()
+				state.EXPECT().PulledContainerMapByArn(taskARN).Return(pulledContainerNameToDockerContainer, true).AnyTimes()
+			} else if tc.name == "Pulled Container Query" {
 				statsEngine.EXPECT().ContainerDockerStats(taskARN, containerID).Return(dockerStats, &stats.NetworkStatsPerSec{}, nil).AnyTimes()
 				state.EXPECT().ContainerMapByArn(taskARN).Return(containerNameToDockerContainer, true).AnyTimes()
 				state.EXPECT().PulledContainerMapByArn(taskARN).Return(pulledContainerNameToDockerContainer, true).AnyTimes()
 				state.EXPECT().TaskByID(containerID).Return(task, true).AnyTimes()
+			} else {
+				statsEngine.EXPECT().ContainerDockerStats(taskARN, containerID).Return(dockerStats, &stats.NetworkStatsPerSec{}, nil).AnyTimes()
+				state.EXPECT().ContainerByID(containerID).Return(dockerContainer, true).AnyTimes()
+				state.EXPECT().ContainerMapByArn(taskARN).Return(containerNameToDockerContainer, true).AnyTimes()
+				state.EXPECT().PulledContainerMapByArn(taskARN).Return(nil, false).AnyTimes()
+				state.EXPECT().TaskByID(containerID).Return(task, true).AnyTimes()
 			}
 			params := graphql.Params{
 				Schema:        schema,
 				RequestString: tc.query,
 				Context:       ctx,
 			}
 			res := graphql.Do(params)
-			if tc.taskContainerError || tc.containerCTXError || tc.taskCTXError || tc.taskByIdError ||
-				tc.dockerStatsError || tc.tagsError || tc.pulledContainerError {
+
+			if tc.taskContainerError || tc.taskByIdError || tc.dockerStatsError || tc.tagsError ||
```

Comment:
nit. Could we split these tests into different sets to have a better readability (and maybe less `if-else` statements)? For example, a set for CTX error, a set for task error, and a set for container error? Thanks!